### PR TITLE
Add script to download latest upgrade image

### PR DESCRIPTION
**Pull request by prakashsurya**

### CODE REVIEW

**Comment by prakashsurya** [comment]
```
$ ./download-latest-image -f internal-dev
download: s3://snapshot-de-images/builds/jenkins-ops/devops-gate/master/appliance-build/master/post-push/373/internal-dev.upgrade.tar.gz to ./internal-dev.upgrade.tar.gz
$ ./download-latest-image internal-dev
download-latest-image: image internal-dev.upgrade.tar.gz already exists
$ ./download-latest-image internal-asdf
fatal error: An error occurred (404) when calling the HeadObject operation: Key "builds/jenkins-ops/devops-gate/master/appliance-build/master/post-push/373/internal-asdf.upgrade.tar.gz" does not exist
download-latest-image: failed downloading image.
```

**Comment by prakashsurya** [comment]
cc: @dtzeng @pandar00 @grwilson @pzakha

File: etc/delphix-platform/upgrade/download-latest-image [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+	f) eval "opt_$c=true" ;;
+	*) usage "illegal option -- $OPTARG" ;;
+	esac
+done
+shift $((OPTIND - 1))
+
+[[ $# -gt 1 ]] && usage "too many arguments specified"
+[[ $# -eq 0 ]] && usage "too few arguments specified"
+
+S3_PREFIX="$(curl --silent "$URL/$JOB/lastSuccessfulBuild/artifact/latest")"
```
Comment (pzakha) [inline etc/delphix-platform/upgrade/download-latest-image]: Any particular reasons for not just getting it from `s3://snapshot-de-images/builds/jenkins-ops/devops-gate/master/appliance-build/master/post-push/latest`?

Comment (prakashsurya) [reply etc/delphix-platform/upgrade/download-latest-image]: That'll probably work; I can make that change.

S3 is "eventually consistent" so I think we could end up downloading a slightly-older-than-latest image, but I don't think that'll be a practical problem.

File: etc/delphix-platform/upgrade/download-latest-image [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+[[ -z "$S3_PREFIX" ]] && die "failed to determine AWS S3 URI"
+
+$opt_f && rm -rf "$1.upgrade.tar.gz" >/dev/null 2>&1
```
Comment (pzakha) [inline etc/delphix-platform/upgrade/download-latest-image]: nit: `-r` is not needed

**Comment by prakashsurya** [comment]
I agree, `/etc` is the wrong place for this, but it's consistently wrong with the `unpack-image` script.. so we can move both with a future change.

**Pull request merged**